src/perftest_resources.c

```diff
@@ -3253,12 +3253,14 @@ void ctx_set_send_reg_wqes(struct pingpong_context *ctx,
 			ctx->wr[i*user_param->post_list + j].wr_id   = i;
 
 			if (j == (user_param->post_list - 1)) {
-				ctx->wr[i*user_param->post_list + j].send_flags = IBV_SEND_SIGNALED;
 				ctx->wr[i*user_param->post_list + j].next = NULL;
+			} else {
+				ctx->wr[i*user_param->post_list + j].next = &ctx->wr[i*user_param->post_list+j+1];
 			}
 
-			else {
-				ctx->wr[i*user_param->post_list + j].next = &ctx->wr[i*user_param->post_list+j+1];
+			if ((j + 1) % user_param->cq_mod == 0) {
+				ctx->wr[i*user_param->post_list + j].send_flags = IBV_SEND_SIGNALED;
```

Comment:
Taking into consideration the comment above, the check should be somewhat like ((i*user_param->post_list + j + 1) % user_param->cq_mod == 0).

Reply:
Please note that `i` is the QP index and the post_lists are separate per QP.